tests/Unit/NumericalAlgorithms/Spectral/Test_SwshCoefficients.cpp

```diff
@@ -0,0 +1,98 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+#include <random>
+#include <sharp_cxx.h>
+
+#include "ErrorHandling/Error.hpp"
+#include "NumericalAlgorithms/Spectral/SwshCoefficients.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+namespace Spectral {
+namespace Swsh {
+namespace {
+
+SPECTRE_TEST_CASE("Unit.NumericalAlgorithms.Spectral.SwshCoefficients",
+                  "[Unit][NumericalAlgorithms]") {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<size_t> sdist{8, 64};
+  const size_t l_max = sdist(gen);
```

Comment:
Why are you drawing this from a random distribution? It is important to have tests succeed or fail deterministically. So if you want to use a random distribution, as a principle you should sample it repeatedly (perhaps O(20) samples) to make sure we don’t occasionally get test failures that are hard to reproduce. Alternatively, you can just test for fixed values, e.g. a small, medium and the max value.

Reply:
Basically every single test  we have does this, both for sizes and for values. I don't see why this test case should be treated differently. We always capture the seed and so it is reproducible.

Reply:
I don't see the point drawing from a random distribution _once_ every time you run the test, as opposed to sampling the distribution or picking fixed values.
Since the seed is captured and failures are therefore reproducible I'd be fine with merging this, but I'm not convinced it's the right strategy.

Reply:
This is literally what we do throughout the entire code base. Something like 90% of our tests do this. I don't see how it's a bad thing. Sampling randomly only once per run keeps the total testing time down and we still get better test coverage over time. This is the recommended method of testing nowadays (fuzzy testing).